include/zephyr/bluetooth/classic/hfp_hf.h

```diff
@@ -53,108 +40,382 @@ struct bt_hfp_hf_cb {
+	 *  status to the application.
+	 *
+	 *  @param call HFP HF call object.
+	 */
+	void (*accept)(struct bt_hfp_hf_call *call);
+	/** HF call reject Callback
+	 *
+	 *  This callback provides the incoming/outgoing call reject
+	 *  status to the application.
+	 *
+	 *  @param call HFP HF call object.
+	 */
+	void (*reject)(struct bt_hfp_hf_call *call);
+	/** HF call terminate Callback
+	 *
+	 *  This callback provides the incoming/outgoing call terminate
+	 *  status to the application.
+	 *
+	 *  @param call HFP HF call object.
+	 */
+	void (*terminate)(struct bt_hfp_hf_call *call);
+	/** HF call held Callback
+	 *
+	 *  This callback provides call held to the application
+	 *
+	 *  @param call HFP HF call object.
 	 */
-	void (*call_held)(struct bt_conn *conn, uint32_t value);
+	void (*held)(struct bt_hfp_hf_call *call);
+	/** HF call retrieve Callback
+	 *
+	 *  This callback provides call retrieved to the application
+	 *
+	 *  @param call HFP HF call object.
+	 */
+	void (*retrieve)(struct bt_hfp_hf_call *call);
 	/** HF indicator Callback
 	 *
 	 *  This callback provides signal indicator value to the application
 	 *
-	 *  @param conn Connection object.
+	 *  @param hf HFP HF object.
 	 *  @param value signal indicator value received from the AG.
 	 */
-	void (*signal)(struct bt_conn *conn, uint32_t value);
+	void (*signal)(struct bt_hfp_hf *hf, uint32_t value);
 	/** HF indicator Callback
 	 *
 	 *  This callback provides roaming indicator value to the application
 	 *
-	 *  @param conn Connection object.
+	 *  @param hf HFP HF object.
 	 *  @param value roaming indicator value received from the AG.
 	 */
-	void (*roam)(struct bt_conn *conn, uint32_t value);
+	void (*roam)(struct bt_hfp_hf *hf, uint32_t value);
 	/** HF indicator Callback
 	 *
 	 *  This callback battery service indicator value to the application
 	 *
-	 *  @param conn Connection object.
+	 *  @param hf HFP HF object.
 	 *  @param value battery indicator value received from the AG.
 	 */
-	void (*battery)(struct bt_conn *conn, uint32_t value);
+	void (*battery)(struct bt_hfp_hf *hf, uint32_t value);
 	/** HF incoming call Ring indication callback to application
 	 *
 	 *  If this callback is provided it will be called whenever there
 	 *  is an incoming call.
 	 *
-	 *  @param conn Connection object.
+	 *  @param call HFP HF call object.
 	 */
-	void (*ring_indication)(struct bt_conn *conn);
-	/** HF notify command completed callback to application
+	void (*ring_indication)(struct bt_hfp_hf_call *call);
+	/** HF call dialing Callback
 	 *
-	 *  The command sent from the application is notified about its status
+	 *  This callback provides call dialing result to the application.
 	 *
-	 *  @param conn Connection object.
-	 *  @param cmd structure contains status of the command including cme.
+	 *  @param hf HFP HF object.
+	 *  @param err Result of calling dialing.
+	 */
+	void (*dialing)(struct bt_hfp_hf *hf, int err);
```

Comment:
Please confirm need use struct bt_hfp_hf_call *call as the input parameter for this callback api?

Reply:
The callback is used to notify the upper layer the result of phone number call/memory dialling call/redial last number.
The call hasn't really started at this point.